src/Runner/Parallel/ProcessPool.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use React\Socket\ServerInterface;
+
+/**
+ * Represents collection of active processes that are being run in parallel.
+ * Inspired by {@see https://github.com/phpstan/phpstan-src/blob/ed68345a82992775112acc2c2bd639d1bd3a1a02/src/Parallel/ProcessPool.php}.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class ProcessPool
```

Comment:
I am super sad that we need to implement it manually, how there could be no available, easy to use solution for this.

Reply:
There's `symplify/easy-parallel` which [has `ProcessPool` too](https://github.com/symplify/easy-parallel/blob/main/src/ValueObject/ProcessPool.php), I just did not want to add another dependency just for that. It's not a big deal, and we're totally free in terms of implementation.